generator/karaf/src/main/java/io/fabric8/maven/generator/karaf/KarafGenerator.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.generator.karaf;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import io.fabric8.maven.core.util.Configs;
+import io.fabric8.maven.core.util.MavenUtil;
+import io.fabric8.maven.docker.config.AssemblyConfiguration;
+import io.fabric8.maven.docker.config.BuildImageConfiguration;
+import io.fabric8.maven.docker.config.ImageConfiguration;
+import io.fabric8.maven.generator.api.BaseGenerator;
+import io.fabric8.maven.generator.api.MavenGeneratorContext;
+import io.fabric8.utils.Strings;
+import org.apache.maven.project.MavenProject;
+
+public class KarafGenerator extends BaseGenerator {
```

Comment:
I know, we don't have yet a good unittest coverage, but I push hard to improve that. Currently we are around 12% (see also https://sonarqube.com/overview?id=io.fabric8%3Afabric8-maven-plugin-build). 

I would be **awesome** if you could provide a unit test for the generator. If you need a mocking framework we use [JMockit](http://jmockit.org/).